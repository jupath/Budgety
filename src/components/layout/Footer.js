import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => (
  <footer className="footer text-center py-4">
    Created by Attila Taligas, view source on <Link to="https://github.com/jupath/Budgety" target="_blank">GitHub</Link>
  </footer>
);

export default Footer;
