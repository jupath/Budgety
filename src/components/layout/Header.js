import React from 'react';

const Header = () => (
  <header>
    <div className="container">
      <h2>Budgety</h2>
      <p>Keep track of your money</p>
    </div>
  </header>
);

export default Header;
