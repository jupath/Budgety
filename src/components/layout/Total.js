import React from 'react';
import { connect } from 'react-redux';
import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';
import itemsSelector from '../../selectors/items';

export const Total = (props) => {
  const { numOfIncomes, numOfExpenses, total } = props;
  const incomeWord = numOfIncomes === 1 ? 'income' : 'incomes';
  const expenseWord = numOfExpenses === 1 ? 'expense' : 'expenses';
  const formedTotal = (<NumberFormat
    value={total}
    displayType="text"
    thousandSeparator
    prefix="$"
    decimalScale={2}
    fixedDecimalScale
  />);
  const renderedTotal = total >= 0 ? formedTotal : <span className="text-danger">{formedTotal}</span>;

  return (
    <h4>
      Viewing {numOfIncomes} {incomeWord} and {numOfExpenses} {expenseWord} totalling {renderedTotal}
    </h4>
  );
};

Total.propTypes = {
  numOfIncomes: PropTypes.number,
  numOfExpenses: PropTypes.number,
  total: PropTypes.number,
};

Total.defaultProps = {
  numOfIncomes: 0,
  numOfExpenses: 0,
  total: 0,
};

const mapStateToProps = state => ({
  numOfIncomes: itemsSelector(state.items, 'income', state.filters).length,
  numOfExpenses: itemsSelector(state.items, 'expense', state.filters).length,
});

export default connect(mapStateToProps)(Total);
