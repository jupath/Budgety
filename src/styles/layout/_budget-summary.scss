.budget-summary {
  background-color: $very-light-gray;
  border-bottom: $border-color 1px solid;
}