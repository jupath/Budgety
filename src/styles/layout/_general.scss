html,
body,
#root,
.site {
  height: 100%;
}

.site {
  display: flex;
  flex-direction: column;
}

body {
  background: url('../images/bg.png') repeat;
  color: $base-font-color;
  font-size: $base-font-size;
}

small {
  font-size: $small-font-size;
}