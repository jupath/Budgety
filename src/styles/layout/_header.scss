.header {
  background: $header-bg-color;
}

.header__content {
  align-content: center;
  color: white;
  display: flex;
  justify-content: space-between;
  line-height: 2.1rem;
  p {
    margin-bottom: 0;
  }
  a:hover {
    text-decoration: none;
  }
}

.header__logo {
  color: white;
  font-family: $font-family-logo;
  font-size: $logo-font-size;
}

.header__logout {
  p {
    @include media-breakpoint-down(sm) {
      font-size: $small-font-size;
    }
  }

  @include media-breakpoint-down(sm) {
    line-height: 1;
  }

  @include media-breakpoint-up(sm) {
    align-items: center;
    display: flex;
  }
}
