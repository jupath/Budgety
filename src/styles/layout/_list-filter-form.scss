.list-filter-form > .container {
  display: flex;
}
