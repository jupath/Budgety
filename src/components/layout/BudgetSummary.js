import React from 'react';
import { connect } from 'react-redux';
import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';
import summarySelector from '../../selectors/summary';
import Total from './Total';

export const BudgetSummary = (props) => {
  const { totalIncomes, totalExpenses } = props;
  const total = totalIncomes - totalExpenses;
  return (
    <div className="budget-summary pt-4 pb-3">
      <div className="container">
        <Total total={total} />
        <h5><span className="pr-2">Total income:</span>
          <NumberFormat
            value={totalIncomes}
            displayType="text"
            thousandSeparator
            prefix="$"
            decimalScale={2}
            fixedDecimalScale
          />
        </h5>
        <h5><span className="pr-2">Total expenses:</span>
          <NumberFormat
            value={totalExpenses}
            displayType="text"
            thousandSeparator
            prefix="$"
            decimalScale={2}
            fixedDecimalScale
          />
        </h5>
      </div>
    </div>
  );
};

BudgetSummary.propTypes = {
  totalIncomes: PropTypes.number,
  totalExpenses: PropTypes.number,
};

BudgetSummary.defaultProps = {
  totalIncomes: 0,
  totalExpenses: 0,
};

const mapStateToProps = state => ({
  totalIncomes: summarySelector(state.items, 'income', state.filters),
  totalExpenses: summarySelector(state.items, 'expense', state.filters),
});

export default connect(mapStateToProps)(BudgetSummary);
