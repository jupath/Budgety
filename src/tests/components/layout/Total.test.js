import React from 'react';
import { shallow } from 'enzyme';
import { Total } from '../../../components/layout/Total';

test('should render Total correctly with default props', () => {
  const wrapper = shallow(<Total />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Total correctly with props', () => {
  const wrapper = shallow(<Total numOfIncomes={2} numOfExpenses={3} total={100} />);
  expect(wrapper).toMatchSnapshot();
});
