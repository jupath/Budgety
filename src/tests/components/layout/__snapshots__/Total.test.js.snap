// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Total correctly with default props 1`] = `
<h4>
  Viewing 
  0
   
  incomes
   and 
  0
   
  expenses
   totalling 
  <NumberFormat
    allowEmptyFormatting={false}
    allowNegative={true}
    decimalScale={2}
    decimalSeparator="."
    displayType="text"
    fixedDecimalScale={true}
    getInputRef={[Function]}
    isAllowed={[Function]}
    isNumericString={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    onValueChange={[Function]}
    prefix="$"
    suffix=""
    thousandSeparator={true}
    type="text"
    value={0}
  />
</h4>
`;

exports[`should render Total correctly with props 1`] = `
<h4>
  Viewing 
  2
   
  incomes
   and 
  3
   
  expenses
   totalling 
  <NumberFormat
    allowEmptyFormatting={false}
    allowNegative={true}
    decimalScale={2}
    decimalSeparator="."
    displayType="text"
    fixedDecimalScale={true}
    getInputRef={[Function]}
    isAllowed={[Function]}
    isNumericString={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    onValueChange={[Function]}
    prefix="$"
    suffix=""
    thousandSeparator={true}
    type="text"
    value={100}
  />
</h4>
`;
